Added the show system menu method to window (#3109)
rust-windowing/winit: CHANGELOG.md

- On Web, add `EventLoopWindowTargetExtWebSys` and `PollStrategy`, which allows to set different strategies for `ControlFlow::Poll`. By default the Prioritized Task Scheduling API is used, but an option to use `Window.requestIdleCallback` is available as well. Both use `setTimeout()`, with a trick to circumvent throttling to 4ms, as a fallback.
- Implement `PartialOrd` and `Ord` for `MouseButton`.
- On X11, fix event loop not waking up on `ControlFlow::Poll` and `ControlFlow::WaitUntil`.

> - added the new `show_system_menu` to `Window` this method can be showed to show system menu even if decoration is disabled. currently only supported on windows.

Reviewer: ```suggestion
- Add `Window::show_system_menu()` which shows the system menu at the mouse position. Currently only implemented on Windows.
```

- On X11, fix event loop not waking up on `ControlFlow::Poll` and `ControlFlow::WaitUntil`.
- **Breaking:** Change default `ControlFlow` from `Poll` to `Wait`.
- **Breaking:** remove `DeviceEvent::Text`.

> - Add `Window::show_window_menu` which shows the system menu at a specified position or the current cursor position. Currently only implemented on Windows.

Reviewer 2: it's very long, something like that should be enough. 

```suggestion
- Add `Window::show_window_menu` to request a titlebar/system menu; implemented on Windows for now. 
```

Reviewer 2: a bit more clear in some cases, that it's the one attached to titlebar. One could think about macOS's app menu integrated into their bar, which is not really what it is.
